zerver/lib/html_diff.py

```diff
@@ -85,9 +100,15 @@ def highlight_html_differences(s1, s2):
     idx = 0
     while idx < len(ops):
         op, text = ops[idx]
-        next_op = None
-        if idx != len(ops) - 1:
-            next_op, next_text = ops[idx + 1]
```

Comment:
I think html_diff originally worked by comparing both the current and next operation (delete / insert / equal) and displaying insertions followed directly by deletions as a single chunk of `replaced` text.